onadata/apps/logger/tests/models/test_xform.py

```diff
@@ -167,6 +168,26 @@ def test_check_xform_uuid(self):
         except DuplicateUUIDError as e:
             self.fail("DuplicateUUIDError raised: %s" % e)
 
+    @mock.patch('onadata.apps.viewer.models.data_dictionary.set_uuid')
+    def test_gen_xform_uuid(self, set_uuid_mock):
+        """
+        Test that an XForms uuid is always set by default on new XForm
+        Objects
+        """
+        self._publish_transportation_form()
+        uuid = self.xform.uuid
+        # The _publish_transportation_form uses the DataDictionary
+        # Model which has some processing steps that ensure XForms
+        # Always have uuids.
+        # Assert that set_uuid is called with an object
+        # that already has a uuid.
+        self.assertEqual(set_uuid_mock.call_count, 1)
```

Comment:
Seems like the `DataDictionary` model always sets uuid for XForms. Sadly, since the `MergedXForm` model inherits from the `XForm` model directly it doesn't have the uuid set...

_I wonder if we should just modify the `save` method of the MergedXForm model to set the uuid ?_

Reply:
Yes, let's have a UUID for the merged form.